python/paddle/fluid/tests/unittests/test_variable.py

```diff
@@ -345,6 +344,52 @@ def _test():
 
         self.assertRaises(Exception, _test)
 
+    def test_size(self):
+        prog = paddle.static.Program()
+        with fluid.program_guard(prog):
+            x = paddle.assign(np.random.rand(2, 3, 4).astype("float32"))
+            exe = fluid.Executor(fluid.CPUPlace())
+            exe.run(fluid.default_startup_program())
+
+            output = exe.run(prog, fetch_list=[x.size()])
+            self.assertEqual(output[0], [24])
+
+    def test_detach(self):
+        b = default_main_program().current_block()
+        x = b.create_var(shape=[2, 3, 5], dtype="float64", lod_level=0)
+        detach_x = x.detach()
+        self.assertEqual(x.persistable, detach_x.persistable)
+        self.assertEqual(x.shape, detach_x.shape)
+        self.assertEqual(x.dtype, detach_x.dtype)
+        self.assertEqual(x.type, detach_x.type)
+        self.assertTrue(detach_x.stop_gradient)
+
+        xx = b.create_var(name='xx', type=core.VarDesc.VarType.STEP_SCOPES)
+        self.assertRaises(AssertionError, xx.detach)
+
+        startup = fluid.Program()
+        main = paddle.static.Program()
+        scope = fluid.core.Scope()
+        with fluid.scope_guard(scope):
+            with fluid.program_guard(main, startup):
+                x = fluid.data(name='x', shape=[3, 2, 1], dtype='float32')
+                x.persistable = True
+                feed_data = np.ones(shape=[3, 2, 1], dtype=np.float32)
+                detach_x = x.detach()
```

Comment:
Could we also test:
1. Change to `detach_x` would also change `x`
2. Change to `x` would also change `detach_x`

Reply:
I have updated code with these tips, thank you.